stablehlo/dialect/TypeInference.cpp

```diff
@@ -3954,6 +4067,56 @@ LogicalResult verifyReduceWindowOp(
   return success();
 }
 
+LogicalResult verifyReshapeOpQuantizationConstraints(
+    std::optional<Location> location, Type operandTy, Type resultTy) {
+  // reshape_c1
+  if (failed(verifyQPerTensorScaleAndZeroPointConstraints(location, operandTy,
+                                                          resultTy)))
+    return failure();
+
+  // reshape_c1
+  if (allQuantized<quant::UniformQuantizedPerAxisType>(operandTy, resultTy)) {
+    if (!isSameQuantPerAxisScaleZeroPoint(operandTy, resultTy)) {
```

Comment:
Let's use the `verify` method you added above. Will cost an extra `if` stmt but should be cheap.